tests/common/fixtures/elasticsearch.py

```diff
@@ -22,7 +22,7 @@ def es_client():
                                 conflicts='proceed')
 
 
-@pytest.fixture
+@pytest.fixture(scope="session", autouse=True)
```

Comment:
There are only a few other places in the code that reference request.es. Mainly the BatchIndexer and cli indexing commands. And they don't really need the whole Client class, they just need the connection.

I don't think it's difficult to mock the default connection, it's just:
```
@pytest.fixture
def mocked_connection():
    client = Mock()
    client.search.return_value = dummy_response()
    connections.add_connection('mock', client)
    yield client
```
And note that the connection is not called 'default'. I think if we did make use of the alias then we would want to pass in the alias to things that use it. So everywhere where you see request.es being passed in we would pass in alias and probably index as well. 

The nice thing about the Client class as it exists today or request.es is that it fully defines the connection to a particular index on a particular cluster. I think it serves an important purpose, otherwise you are passing around the index and alias everywhere. And it's nice to know in order to make a different connection all you have to do is create another instance of the Client class. We could still take advantage of the aliasing by using elasticsearch dsl to reference the connection in the Client class. At the end of the day I really don't think it matters either way, whether we create the client via elasticsearch_dsl or via elasticsearch and pass it to elasticsearch_dsl; elasticsearch_dsl excepts both an alias or an elasticsearch client connection as an argument.

Reply:
As you noted - the existing `Client` class encapsulates a bunch of useful things (connection, index). I also like the fact that you can easily trace how it gets initialized and set up because it works in a way that is consistent with other aspects of the code too (eg. `request.db`). The elasticsearch_dsl connection depends on a hidden global inside elasticsearch_dsl.

What I would definitely suggest is that to make this PR safer to deploy, we continue to use the same elasticsearch connection but just change the query construction code.